api-2.2/src/main/java/org/openmrs/module/reporting/cohort/definition2_2/evaluator2_2/ConditionCohortDefinitionEvaluator.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.reporting.cohort.definition2_2.evaluator2_2;
+
+import java.util.List;
+
+import org.openmrs.Cohort;
+import org.openmrs.Condition;
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.reporting.cohort.EvaluatedCohort;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.evaluator.CohortDefinitionEvaluator;
+import org.openmrs.module.reporting.cohort.definition2_2.ConditionCohortDefinition;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.querybuilder.HqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Handler(supports = { ConditionCohortDefinition.class })
+public class ConditionCohortDefinitionEvaluator implements CohortDefinitionEvaluator {
+	
+	@Autowired
+	EvaluationService evaluationService;
+	
+	@Override
+	public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationContext context) {
+		
+		ConditionCohortDefinition cd = (ConditionCohortDefinition) cohortDefinition;
+		
+		HqlQueryBuilder query = new HqlQueryBuilder();
+		query.select("c.patient.patientId");
+		query.from(Condition.class, "c");
+		query.wherePatientIn("c.patient.patientId", context);
+		
+		if (cd.getConditionCoded() != null) {
```

Comment:
All of these null checks are unnecessary.  If you look at how the HqlBuilder is implemented, it will check for null and do nothing if a null is passed in.  So you can clean up this code by removing all 4 null checks.

Reply:
cleaned up the unnecesary null checks